ts/screens/wallet/TransactionDetailsScreen.tsx

```diff
@@ -102,19 +108,49 @@ const contextualHelpMarkdown: ContextualHelpPropsMarkdown = {
  */
 class TransactionDetailsScreen extends React.Component<Props> {
   public componentDidMount() {
-    BackHandler.addEventListener(
-      "hardwareBackPress",
-      this.props.navigateToWalletHome
-    );
+    BackHandler.addEventListener("hardwareBackPress", this.handleBackPress);
   }
 
   public componentWillUnmount() {
-    BackHandler.removeEventListener(
-      "hardwareBackPress",
-      this.props.navigateToWalletHome
-    );
+    BackHandler.removeEventListener("hardwareBackPress", this.handleBackPress);
   }
 
+  private getRouteName = (route: NavigationRoute): any => {
```

Comment:
I guess this code could be more string if we use Option

Note: if you find it as a valid solution, make some tests because I didn't test this code
```typescript
private getRouteName = (route: NavigationRoute): Option<string> => {
    if (route.index === undefined) {
      return some(route.routeName);
    }
    if (route.index >= route.routes.length) {
      return none;
    }
    return this.getRouteName(route.routes[route.index]);
  };

  private whereAmIFrom = (): Option<string> => {
    if (this.props.nav.length === 0) {
      return none;
    }
    const navLength = this.props.nav.length;
    const lastStep = this.props.nav[navLength - 1].routes[
      this.props.nav[navLength - 1].index
    ];
    return this.getRouteName(lastStep);
  };

  // On back button navigate to wallet home (android)
  private handleBackPress = () => {
    if (this.whereAmIFrom().fold(false, r => r === "WALLET_HOME")) {
      return this.props.navigation.goBack();
    } else {
      this.props.navigateBackToEntrypointPayment();
      return true;
    }
  };
```